test/complex_boolean_test.cpp

```diff
@@ -0,0 +1,1026 @@
+// Copyright 2021 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "manifold.h"
+#include "polygon.h"
+#include "test.h"
+
+using namespace manifold;
+
+/**
+* These tests verify correct topology and geometry for complex boolean
+* operations between valid shapes with many faces.
+*/
+
+TEST(Boolean, Sphere) {
+ Manifold sphere = Manifold::Sphere(1.0f, 12);
```

Comment:
It looks like our formatter is mad because you have one-space tabs instead of two. We have a .clang-format file that should set everything up correctly if you just run clang-format (I usually use the VSCode extension).

Reply:
Done thanks!